common/src/chain/transaction/signature/mod.rs

```diff
@@ -294,8 +297,15 @@ fn stream_signature_hash<T: Signable>(
     inputs.signature_hash(stream, mode, target_input, target_input_num)?;
     outputs.signature_hash(stream, mode, target_input, target_input_num)?;
 
-    // TODO: consider doing just like taproot, and hash in all outputs that come from the outpoints of inputs,
-    //       this would be a good solution to avoid having to download full transactions to verify inputs
+    // Include utxos of the inputs to make it possible to verify the inputs scripts and amounts without downloading the full transactions
+    if inputs.len() != inputs_utxos.len() {
+        return Err(TransactionSigError::InvalidUtxoCountVsInputs(
+            inputs_utxos.len(),
+            inputs.len(),
+        ));
+    } else {
+        hash_encoded_to(&inputs_utxos, stream);
```

Comment:
I don't think this will play nicely with the likes of `SIGHASH_ANYONECANPAY`. This basically always commits to all inputs, regardless of the flag.

From wiki:
> The SIGHASH_ANYONECANPAY modifier can be combined with the above three modes. When set, only that input is signed and the other inputs can be anything.

Maybe the spent output commitment could be a part of tx input sighash calculation, so it shares the ANYONECANPAY logic + you get utxo being spent and the corresponding input nicely next to each other.

Reply:
This isn't compatible with anyonecanpay. Neither was the previous version. That'll will be done later.